src/SessionData.php

```diff
@@ -12,43 +11,24 @@
  */
 class SessionData implements Session
 {
-    /**
-     * @var string Client Session ID (UUID)
-     */
-    private $client_session_id;
 
     /**
      * @var string|null old Session ID (if this Session was renewed)
      *
      * @see renew()
      */
-    private $old_session_id;
-
-    /**
-     * @var mixed[] map where fully-qualified class-name => [checksum, serialized data]
-     */
-    private $data = [];
+    private ?string $old_session_id = null;
 
     /**
      * @var SessionModel[] map where fully-qualified class-name => instance
      */
-    private $objects = [];
+    private array $objects = [];
 
-    /**
-     * @var bool indicates whether this is a new Session (for which no Session Cookie has been set yet)
-     */
-    private $is_new = false;
-
-    /**
-     * @param string $client_session_id
-     * @param array  $data
-     * @param bool   $is_new
-     */
-    public function __construct(string $client_session_id, array $data, bool $is_new)
-    {
-        $this->client_session_id = $client_session_id;
-        $this->data = $data;
-        $this->is_new = $is_new;
+    public function __construct(
+        private string $client_session_id,
+        private array $data,
+        private bool $is_new = false
```

Comment:
```suggestion
        private bool $is_new
```

Adding it as a default is backwards incompatible

Reply:
It already has a default in line 40. I am not sure how far backwards compatible we are talking about. composer.json says minimum php 8.0 and everything passes in 8.0 environment.

Reply:
It wasn't default in the constructor signature, which is the breaking part :-)

original: `public function __construct(string $client_session_id, array $data, bool $is_new)`

Reply:
Because it is public, we should stay strict - we can cut corners on private repositories because we know how we extend and when a b\c break is an issue or not.